### ISSUE TITLE
SmallRye Metrics - @Gauge methods analyzed in MetricsInterceptor should be registered for reflection

### ISSUE THREAD

**Issue by mkouba** [quarkusio/quarkus#6993]
**Describe the bug**
`MetricsInterceptor` does not see the methods annotated with `@Gauge` if they're not registered for reflection by another extension or annotated with `@RegisterForReflection`.

**Expected behavior**
Methods are registered automatically.

**Actual behavior**
Methods are not registered automatically. `MetricsInterceptor` does not register a gauge method. No error or warning message.

**To Reproduce**
1. Add the following bean class to the `microprofile-metrics-quickstart`
2. Build the native image
3. Run the image - you should see the `STARTUP` message in the log but the gauge is not listed on http://localhost:8080/metrics/application

```java
@ApplicationScoped
public class DummyGauge {

    void init(@Observes StartupEvent event) {
       System.out.println("STARTUP!");
    }

    @Gauge(name = "dummyGauge", unit = MetricUnits.NONE)
    public Long dummyGauge() {
        return 42l;
    }
}
```

**Workaround**
Annotated the class with `@RegisterForReflection`.

**Comment by geoand** [comment]
Will this also be solved by #6989?

**Comment by mkouba** [comment]
Nope, that's a different issue. This registration must be perfomed in the SM Metrics extension.
